lib/plugins/digging.js

```diff
@@ -161,7 +161,7 @@ function inject (bot) {
       bot._client.write('block_dig', {
         status: 1, // cancel digging
         location: bot.targetDigBlock.position,
-        face: 1 // hard coded to always dig from the top
+        face: 0 // hard coded to always dig from the bottom
```

Comment:
> Is this correct for all versions?

 I'm not sure. The protocol wiki doesn't show any info about it on older versions than 1.20.4 (it's only in 1.20.4 because i just added it) https://wiki.vg/Protocol
 
Does anyone know a proper way to test this for every version mineflayer supports? Manually setting up a server and launching client for every version will take a very long time.

Reply:
You mention "On a server i noticed the block often doesnt break anymore when breaking something with hand again after cancelling. With this change this issue doesnt seem to happen so it possibly triggered anticheat or some glitch before" is what this PR is fixing

There is 2 ways to test this
1. You implement that as an external test in mineflayer so it will run in every version
2. You test it manually for at least a few version like 1.8 and 1.13 and 1.20.4

Is there a need for the vanilla client for this text ?

Reply:
> Is there a need for the vanilla client for this text ?
I think so or at least not mineflayer, the goal is to find out what value the vanilla client sends for the facing and it doesn't seem to be properly documented anywhere. My assumption is that it's hardcoded to be facing 0 for any other version too (instead of the current hardcoded 1 in mineflayer).

I don't think the first way of testing is practical to find out the vanilla value. It will most likely work with any facing value on most servers regardless of version, it probably doesn't get used.
It made no difference if it's 1 or 0 either in my local paper test server without anticheat. The server where the wrong diggingface (1) did gave issues presumably has matrix anticheat, which advertises with verifying packets / values.

I guess the only way to find the vanilla value is to sniff the packets of other vanilla client versions.

@zardoy How did you get the info about hardcoded diggingface 0? Did you manually sniff the packets too? What version was that?

Reply:
@Vinciepincie Im using pakkit with 1.18.2 but it's better to use the tool you linked since it doesn't use Minecraft node protocol. Also you don't need to install a server for each version you can just find a server with multiple versions support or use a proxy locally (most of them support 1.8- latest)

Reply:
update:
I used the pakkit tool for sniffing packets instead, sniffcraft doesnt work for me for other versions than 1.20.4..
[](https://github.com/Heath123/pakkit)

I noticed something new. 

When holding left click and moving around multiple blocks the start and cancellation packet gets triggered but the cancellation face stays the same as the dig face. Only the last block where you release left click will be hardcoded to 0.

I didn't have time to test all the versions yet but the output is the same so far. So far I tried 1.8.9, 1.19.5 and 1.20.4. 
I will continuing testing more versions later. Should this pr be converted to draft?


Here's the output.
```
// start break
{
status:
0,
location:
{ x: -190, z: 266, y: 64 },
face:
1,
sequence:
15
}

// cancel, but still holding left click, moving to other block.
{
status:
1,
location:
{ x: -190, z: 266, y: 64 },
face:
1,
sequence:
0
}

// start breaking new block
{
status:
0,
location:
{ x: -189, z: 266, y: 64 },
face:
1,
sequence:
16
}

// cancel digging new block
{
status:
1,
location:
{ x: -189, z: 266, y: 64 },
face:
0,
sequence:
0
}
```

Also the (anticheat?) issue on the specific server still sometimes appear, but less than before with the hardcoded 1.

Reply:
I tested moving from block to another without breaking, and releasing left click on the second block.


Cancellation due to cursor moving to other block (holding left click):
First start and cancel packets will have the same digingface. 

Cancellation because of releasing left click or other reasons:
Cancel packet will always have hardcoded 0 as diggingface.

This behaviour is the same for all minecraft vanilla clients between 1.8 - 1.20


https://gist.github.com/Vinciepincie/2eb8db547e2cb7a3f06fc76d22b37bba